### ISSUE TITLE
self doesn't need to be mutable lint

### ISSUE THREAD

**Issue by Detector-I** [rust-lang/rust-clippy#9591]
### What it does

Having a function with `&mut self` as an argument, and actually never use `self` as mutable inside it wont give us any compiler or clippy warning...
here is an example
```rs
fn method(&mut self) -> String {
    self.a_str.to_string()
}
```
I think in this case clippy should say `self doesn't need to be mutable`.

### Lint Name

unused_mutable_self

### Category

correctness

### Advantage

- self wont be borrowed anymore in this function when its dont needed

### Example

```rust
fn method(&mut self) -> String {
    self.a_str.to_string()
}
```

Could be written as:

```rust
fn method(&self) -> String {
    self.a_str.to_string()
}
```

**Comment by kraktus** [comment]
Same as https://github.com/rust-lang/rust-clippy/issues/5546 ?
